build.sh

```diff
@@ -2,28 +2,39 @@
 
 TAG="$(< VERSION)-$(git rev-parse --short HEAD)"
 
-echo "Getting updated images"
-docker-compose pull conjur_4 conjur_5
+echo "Getting updated images (this may take a few minutes)..."
+docker-compose pull -q conjur_4 conjur_5
 
 echo "Building Buildpack Health Check executable"
 rm -rf bin/buildpack-health-check
 docker-compose -f buildpack-health-check/docker-compose.yml build
 docker-compose -f buildpack-health-check/docker-compose.yml \
   run --rm buildpack-health-check-builder
 
-echo "Building conjur-service-broker Docker image"
-docker-compose build conjur-service-broker
-docker-compose build tests
+echo "Building conjur-service-broker image"
+docker build -t "conjur-service-broker:$TAG" \
+  -t "conjur-service-broker:latest" \
+  .
 
-echo "Tagging conjur-service-broker:$TAG"
-docker tag conjur-service-broker "conjur-service-broker:$TAG"
+echo "Building conjur-service-broker-test image"
+docker build -t "conjur-service-broker-test:latest" \
+  -f Dockerfile.test \
+  .
 
 echo "Running deployment build to install dependencies locally"
 echo "Creating project ZIP file"
-docker-compose run \
-  --rm \
+docker run --rm \
+  -v "$(pwd):/output" \
   conjur-service-broker \
-  bash -c "
-    bundle pack --all
-    zip -r cyberark-conjur-service-broker_$(cat VERSION).zip ./*
+  bash -ec "
+    echo 'Bundling dependencies...'
+    bundle package --all --no-install
+
+    echo 'Removing the old zip...'
+    rm -rf /output/cyberark-conjur-service-broker_$(cat VERSION).zip
+
+    echo 'Zipping everything up...'
+    zip -r \
+      /output/cyberark-conjur-service-broker_$(cat VERSION).zip ./* \
+      -x vendor/bundle
```

Comment:
what does the `-x` do?

Reply:
Excludes the directory. `vendor/bundle` is where the gems are installed.

Reply:
so the zip doesn't include the packaged gems?

it is a requirement that the ZIP includes the vendored gems so that customers installing the service broker from an airgapped environment can still install the service broker and all of its runtime dependencies appropriately

Reply:
> so the zip doesn't include the packaged gems?

It's a bit complicated (as I found out myself). You have:
- Bundle Cache (`vendor/cache`): This is where the sources of those gems are stored. If something is needed to be installed, this directory will substitute for rubygems.
- Bundle installation directory (`vendor/bundle`): This is where those gems are installed. This is needed in our dockerfile since it needs to run `rails`.

You can use the cache to reconstruct the installation (with a few caveats that you can find [here](https://bundler.io/v1.6/bundle_package.html))

tl;dr: The gems will be installable it's just that we are only including the source package rather than also including the "installed" versions of those packages.

Reply:
ok, that sounds reasonable. what is tricky is finding a way to test this :) maybe I should try running the service broker in docker without the internet on and see if it works?

Reply:
ran it with my internet off, and the tests passed.